plugins/base/src/main/kotlin/renderers/PackageListService.kt

```diff
@@ -31,27 +34,27 @@ class PackageListService(val context: DokkaContext, val rootPage: RootPageNode)
                     ?: run { context.logger.error("Cannot resolve path for ${node.name}!"); null }
 
                 if (dri != DRI.topLevel && locationProvider.expectedLocationForDri(dri) != nodeLocation) {
-                    nonStandardLocations[dri.toString()] = "$nodeLocation.$linkExtension"
+                    nonStandardLocations[dri.toString()] = "$nodeLocation.${format.linkExtension}"
                 }
             }
 
             node.children.forEach { visit(it) }
         }
 
         visit(module)
+        return renderPackageList(nonStandardLocations, mapOf("" to packages), format.formatName, format.linkExtension)
+    }
 
-        return buildString {
+    companion object {
+        fun renderPackageList(nonStandardLocations: Map<String, String>, modules: Map<String, Set<String>>, format: String, linkExtension: String): String = buildString {
             appendLine("$DOKKA_PARAM_PREFIX.format:${format}")
             appendLine("$DOKKA_PARAM_PREFIX.linkExtension:${linkExtension}")
             nonStandardLocations.map { (signature, location) -> "$DOKKA_PARAM_PREFIX.location:$signature\u001f$location" }
-                .sorted().joinTo(this, separator = "\n", postfix = "\n")
+                    .sorted().joinTo(this, separator = "\n", postfix = "\n")
 
-            packages.sorted().joinTo(this, separator = "\n", postfix = "\n")
+            modules.map { (module, packages) ->
+                "$MODULE_DELIMITER$module\n".takeIf { module.isNotBlank() }.orEmpty() + packages.filter(String::isNotBlank).sorted().joinToString(separator = "\n")
```

Comment:
Is this a problem that we could have just the module name (without packages) or just packages (without module)

Reply:
Packages without module can't happen, empty modules fixed